### PR TITLE
[ENH] Thuml Sundial forecasting foundation model

### LINKED ISSUE #10419: [ENH] Add support for Thuml's Sundial
**Model:** Sundial by Thuml
**Downloads:** 109,674
**License:** Apache-2.0
**Links:**
- HF weights: https://huggingface.co/thuml/sundial-base-128m
- Paper: Sundial: A Family of Highly Capable Time Series Foundation Models, https://arxiv.org/abs/2502.00816

**Implementation notes:**

- Released by Thuml in March 2026.
- The modeling code needs to be forked in `sktime/libs`
- After that, it can be implemented like other HuggingFace based models.

### PR DESCRIPTION
**Pull request by geetu040**
Fixes #10419

Adds `SundialForecaster`, a new `sktime` interface for Thuml’s Sundial model using a local `transformers` implementation.

Changes:

* Add zero-shot forecasting support for Sundial
* Add quantile prediction support from generated sample paths
* Support `device_map`, `dtype`, and generation options via `forward_kwargs`
* Add deterministic generation option
* Add docs, examples, and VM test support
* Vendor the Sundial model implementation with upstream Apache-2.0 attribution

CC: @WenWeiTHU

### CODE REVIEW

**Review by fkiraly — CHANGES_REQUESTED** [review]
Nice! Quick question, generating sample paths is usually a tedious way to generate quantile forecasts. Have you checked whether the quantile forecasts can be generated directly?

**Comment by geetu040** [comment]
> Nice! Quick question, generating sample paths is usually a tedious way to generate quantile forecasts. Have you checked whether the quantile forecasts can be generated directly?

@fkiraly I checked the Sundial implementation and couldn't find a way to obtain quantiles directly. The model exposes the sampling path and returns raw logits/samples, but there doesn't appear to be any internal or public API for direct quantile prediction. For now, quantiles are computed empirically from the generated samples.

**Review by fkiraly — APPROVED** [review]
